src/plugins/GUITestBase/src/GTUtilsDashboard.cpp

```diff
@@ -146,6 +207,45 @@ bool GTUtilsDashboard::hasNotifications(HI::GUITestOpStatus &os) {
     return notificationsWidget->isVisible();
 }
 
+#define GT_METHOD_NAME "getNotifications"
+QList<GTUtilsDashboard::Notification> GTUtilsDashboard::getNotifications(GUITestOpStatus &os) {
+    using Type = Notification::Type;
+
+    const QString notificationsWidgetName = QStringLiteral("NotificationsDashboardWidget");
+    const auto tableLayout = qobject_cast<QGridLayout *>(
+        GTWidget::findWidget(os, notificationsWidgetName, GTUtilsDashboard::getDashboard(os))->layout());
+    QList<Notification> notifications;
+
+    GT_CHECK_RESULT(tableLayout != nullptr && tableLayout->columnCount() == 3,
+                    notificationsWidgetName % QStringLiteral(" was found, but cannot be used in a test"),
+                    notifications);
+
+    const int notificationsNumber = tableLayout->rowCount();
+    notifications.reserve(notificationsNumber);
+    for (int row = 0; row < notificationsNumber; ++row) {
+        const QString typeStr = getNotificationCellText(os, *tableLayout, row, 0);
+        const QString element = getNotificationCellText(os, *tableLayout, row, 1);
+        const QString message = getNotificationCellText(os, *tableLayout, row, 2);
+        if (!typeStr.isNull() && !element.isNull() && !message.isNull()) {
+            const QVariant type = getNotificationTypeFromString(typeStr);
+            if (!type.isNull()) {
+                notifications << Notification {static_cast<Type>(type.toInt()), element, message};
+            }
+        }
+    }
+    return notifications;
+}
+#undef GT_METHOD_NAME
+
+QString GTUtilsDashboard::getJoinedNotificationsString(GUITestOpStatus &os) {
+    const auto notifications = getNotifications(os);
+    QStringList stringNotifications;
+    for (const auto &notification : notifications) {
+        stringNotifications << notification;
```

Comment:
I would avoid operator overloading completely and called toString() here to make the code more readable and less dependable from the external context.